amt/repositories/tasks.py

```diff
@@ -19,26 +19,28 @@ class TasksRepository:
     The TasksRepository provides access to the repository layer.
     """
 
-    def __init__(self, session: Annotated[Session, Depends(get_session)]) -> None:
+    def __init__(self, session: Annotated[AsyncSession, Depends(get_session)]) -> None:
         self.session = session
 
-    def find_all(self) -> Sequence[Task]:
+    async def find_all(self) -> Sequence[Task]:
         """
         Returns all tasks in the repository.
         :return: all tasks in the repository
         """
-        return self.session.execute(select(Task)).scalars().all()
+        result = await self.session.execute(select(Task))
+        return result.scalars().all()
 
-    def find_by_status_id(self, status_id: int) -> Sequence[Task]:
+    async def find_by_status_id(self, status_id: int) -> Sequence[Task]:
         """
         Returns all tasks in the repository for the given status_id.
         :param status_id: the status_id to filter on
         :return: a list of tasks in the repository for the given status_id
         """
         statement = select(Task).where(Task.status_id == status_id).order_by(Task.sort_order)
-        return self.session.execute(statement).scalars().all()
+        result = await self.session.execute(statement)
+        return result.scalars().all()
```

Comment:
To avoid using a variable (result) and then return, you could consider combining such lines to:
```suggestion
        return (await self.session.execute(statement)).scalars().all()
```

Reply:
Yes nice